.github/workflows/windows_cmake.yml

```diff
@@ -0,0 +1,163 @@
+        echo "LIBT_BSL=OFF" >> $GITHUB_ENV
+        echo "VCPKG_TRIPLET=x64-windows-static" >> $GITHUB_ENV
+        echo "BUILD_VARIANT=static" >> $GITHUB_ENV
+
+    - name: setup environment - don't build with deprecated functions
+      if: matrix.deprecated_functions == 'no_depr_fun'
+      shell: bash
+      run: |
+        echo "LIBT_DEPR_FUN=OFF" >> $GITHUB_ENV
+
+    - name: setup environment - don't build tests
+      if: matrix.build_tests == 'no_tests'
+      shell: bash
+      run: |
+        echo "LIBT_TESTS=OFF" >> $GITHUB_ENV
+
+    - name: setup environment - don't build with python bindings
+      if: matrix.python_bindings == 'no_py_bindings'
+      shell: bash
+      run: |
+        echo "LIBT_PY_BINDINGS=OFF" >> $GITHUB_ENV
+
+    # NOTE: MSVC tools must be in the path for the Ninja generator. Caveat: must use cmd
+    - name: setup MSVC dev cmd
+      uses: ilammy/msvc-dev-cmd@v1.4.1
+
+    # ninja is preferrable, not a hard requirement
+    - name: install additional required packages with chocolatey
+      run: |
+        choco install ninja
+
+    - name: setup vcpkg (cached, if possible)
+      uses: lukka/run-vcpkg@v4.1
+      with:
+        vcpkgDirectory: ${{ env.VCPKG_DEST_WIN }}
+        vcpkgGitCommitId: ${{ env.VCPKG_COMMIT }}
+        setupOnly: true
+        appendedCacheKey: ${{ matrix.build_variant }}
+
+    # clear buildtrees to reduce disk space requirements
+    - name: install dependencies via vcpkg
+      run: |
+        ${{ env.RUNVCPKG_VCPKG_ROOT }}/vcpkg.exe install `
+          boost-asio:${{ env.VCPKG_TRIPLET }} `
+          boost-chrono:${{ env.VCPKG_TRIPLET }} `
+          boost-config:${{ env.VCPKG_TRIPLET }} `
+          boost-crc:${{ env.VCPKG_TRIPLET }} `
+          boost-date-time:${{ env.VCPKG_TRIPLET }} `
+          boost-iterator:${{ env.VCPKG_TRIPLET }} `
+          boost-multiprecision:${{ env.VCPKG_TRIPLET }} `
+          boost-pool:${{ env.VCPKG_TRIPLET }} `
+          boost-python:${{ env.VCPKG_TRIPLET }} `
+          boost-random:${{ env.VCPKG_TRIPLET }} `
+          boost-scope-exit:${{ env.VCPKG_TRIPLET }} `
+          boost-system:${{ env.VCPKG_TRIPLET }} `
+          boost-variant:${{ env.VCPKG_TRIPLET }} `
+          openssl:${{ env.VCPKG_TRIPLET }} `
+          --clean-after-build
+
+    - name: build libtorrent
+      shell: cmd
+      run: |
+        cmake -B cmake-build-dir -G "Ninja" ^
+          -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DCMAKE_BUILD_TYPE=${{ matrix.build_config }} ^
+          -DCMAKE_TOOLCHAIN_FILE=${{ env.VCPKG_DEST_WIN }}\scripts\buildsystems\vcpkg.cmake ^
+          -DVCPKG_TARGET_TRIPLET=${{ env.VCPKG_TRIPLET }} ^
+          -Dbuild_examples=ON -Dbuild_tools=ON ^
+          -Dbuild_tests=${{ env.LIBT_TESTS }} -Ddeprecated-functions=${{ env.LIBT_DEPR_FUN }} ^
+          -DBUILD_SHARED_LIBS=${{ env.LIBT_BSL }} -Dstatic_runtime=${{ env.LIBT_STATIC_RT }} ^
+          -Dpython-bindings=${{ env.LIBT_PY_BINDINGS }} -Dskip-python-runtime-test=ON ^
```

Comment:
@arvidn 

> it seems a bit unfortunate to introduce these environment variables (since it increases complexity a bit). If I understand correctly, the reason you can't just use `${{ matrix.python_bindings }}` here, is because then you would need to name those options `YES` and `NO` in the build matrix. And the reason you don't want to do that is because you end up with an artifact named `-YES-NO-NO-YES` and it becomes cryptic. Is that right?

That's exactly right

> Part of me feels like the complexity could be contained a bit better if it was concentrated around generating a legible name for the artifact.

Yeah, I suppose I should be able to push all the complexity down to a single step dedicated to building the artifact name. I'll see what I can do.